building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+
+
+Before proceeding, create a _.gitignore_ file in the project folder to prevent checking in the "venv" folder to git:
+
+
+```
+$ touch .gitignore
+```
+
+Add the following:
+
+
+```
+venv
+__pycache__
+```
+
+With the necessary installations in place, log on to Okteto from your terminal using the command:
+
+```sh
+$ okteto login
+```
+
+After logging in, run `okteto init` to create an Okteto manifest file. The command automatically generates an `okteto.yml` file. Overwrite the content to:
+
+```yaml
+name: fastapi-okteto
+command: bash
+workdir: /app
+```
+
+The next step is to create a stack manifest file for Okteto. The manifest file removes the need to deal with the complexities of Kubernetes manifests.
+
+```
+$ touch okteto-stack.yaml
+```
+
+In the `okteto-stack.yaml` file, define the application using a docker-compose format:
+
+```yaml
+name: fastapi-crud
+services:
+  fastapi:
+    public: true
+    image: fastapi-crud:latest
+    build: .
+    replicas: 1
+    ports:
+      - 8080
+    resources:
+      cpu: 100m
+      memory: 128Mi
+```
+
+In the manifest file above, you defined your application's name and the services your app is made up of, fastapi. The fastapi service exposes the port `8080` and allocates itself with 100 mili CPUs and 128Mi of memory. 
+
+> Okteto has a reference page on [https://okteto.com/docs/reference/stacks](Okteto Stacks Manifest).
+
+With the manifest in place, create a Dockerfile to house the build instruction for the application. Dockerfile contains build instruction for images to be deployed on Okteto.
+
+```sh
+$ touch Dockerfile
+```
+
+Add the following:
+
+```Dockerfile
+FROM python:3.8
+
+ADD requirements.txt /requirements.txt
+
+ADD main.py /main.py
+
+ADD okteto-pipeline.yaml /okteto-pipeline.yaml
+
+ADD okteto-stack.yaml /okteto-stack.yaml
+
+ADD okteto.yml /okteto.yml
+
+RUN pip install -r requirements.txt
+
+EXPOSE 8080
+
+COPY ./app app
+
+CMD ["python3", "main.py"]
+```
+
+Next, create an `okteto-pipeline.yaml` file. The Okteto pipeline file gives you more control over how you want to deploy and build your application. Add the following in the pipeline file:
+
+```yaml
+deploy:
+  - okteto stack deploy --build
+dev:
+  - okteto.yml
+```
+
+In the pipeline file above, you defined the instruction for deployment and development. When Okteto deploys your application, it references the instruction listed in the stack manifest, and when it's developing the application, it references the Okteto manifest for instructions.
+
+Before proceeding to deployment on Okteto, download and set up your Kubernetes credentials on your local machine:
+
+![Okteto Dashboard](https://res.cloudinary.com/adeshina/image/upload/v1608635743/x9tmv4styid2hffxilxx.png)
+
+Let's deploy your application to Okteto. Start by running the command: 
+
+```
+okteto namespace
+```
+
+The command above setups a namespace on the Okteto cloud. Next, run the command:
+
+```
+okteto up
```

Comment:
instead of this command, I think we should use `okteto stack deploy --build`, since we are not doing any "live coding". We can explain that running that command automatically builds your service using Okteto's build service, and then deploys the application.

Reply:
I get this error when I run the command you gave:

```
error building image for 'fastapi': build failed: failed to solve: rpc error: code = Unknown desc = server message: insufficient_scope: authorization failed
```

Reply:
I've fixed this. I should add that the user has an active docker hub session on his/her machine.